interface1.js

```diff
@@ -884,35 +884,200 @@ window.addEventListener('load', function(){
 			genStringsList = fixedStringList;
 		}
 
+		/* minProblem, maxProblem, and inputProblem are flags for specific tests. The default value is 0. If the value equals to 1, then the test failed */
+		var minProblem = 0; var maxProblem = 0;
+		var inputProblem = 0;
 		var length = spotForm.genStringsInput.length;
 		if(length === undefined) {
 			length = 1;
 		}
-		var inputList = spotForm.genStringsInput.value;
-		var min = spotForm.genStringsMin.value;
-		var max = spotForm.genStringsMax.value;
 
+		/* checking if list of terminals input is empty */
 		for(var i=0; i<length; i++){
-			if(length > 1) {
-				inputList = spotForm.genStringsInput[i].value;
-				min = spotForm.genStringsMin[i].value;
-				max = spotForm.genStringsMax[i].value;
+			if (length > 1){
+				if (spotForm.genStringsInput[i].value !== ""){
+					inputProblem = 1;
+				}
+			}else{
+				if (spotForm.genStringsInput.value !== ""){
+					inputProblem = 1;
+				}
+			}
+		}
+		if(inputProblem == 1){
+			/* checking if min or max is empty */
+			for(var i=0; i<length; i++){
+				if (length > 1){
+					if (spotForm.genStringsInput[i].value !== ""){
+						if (spotForm.genStringsMin[i].value === ""){
+							minProblem = 1;
+						}
+						if (spotForm.genStringsMax[i].value === ""){
+							maxProblem = 1;
+						}
+					}
+				}else{
+					if (spotForm.genStringsMin.value === ""){
+						minProblem = 1;
+					}
+					if (spotForm.genStringsMax.value === ""){
+						maxProblem = 1;
+					}
+				}
 			}
+			if (minProblem == 0 && maxProblem == 0){	
+				/* checking if the input is a number */
+				for(var i=0; i<length; i++){
+					if (length > 1){
+						if (spotForm.genStringsInput[i].value !== ""){
+							if (isNaN(spotForm.genStringsMin[i].value)){
+								minProblem = 1;
+							}
+							if (isNaN(spotForm.genStringsMax[i].value)){
+								maxProblem = 1;
+							}
+						}
+					}else{
+						if (isNaN(spotForm.genStringsMin.value)){
+							minProblem = 1;
+						}
+						if (isNaN(spotForm.genStringsMax.value)){
+							maxProblem = 1;
+						}
+					}
+				}
+				if (minProblem === 0 && maxProblem === 0){
```

Comment:
I guess the reason for the current organization is to be able to give more informative errors.

How about flipping the logic of the conditionals, so that instead of nesting them all, it goes:
if(input is non empty but min is empty or max is empty){minOrMaxProblem = 1; display appropriate error}
elseif(isNaN(min) or isNaN(max)){minOrMaxProblem = 1; display a different appropriate error}
elseif(other checks...)
else //if you get here, there are no problems with min or max

But perhaps I'm overlooking something and the current organization is unavoidable.

Reply:
Yes, the current organization was to be able to give more informative errors. I will try the suggestion, I think that will work better. Thanks!

Reply:
Sounds good!

Reply:
After rethinking, I am not sure if that will work because there needs to be a for loop before each statement that checks all the string. On second thought, I think I structure my code like that previously but that only work for the first input and it didn't work for input 2 and onward so I switched it to the current structure. I will try to rethink about it again though.

Reply:
Hmm. I was thinking that all of the checks could be in one for-loop. But maybe i'm missing something. You can let me know tomorrow.

Reply:
I figured out how to do it in one for-loop. I think it's less complicated and less code.